scripts/tests/scripts_regression_tests.py

```diff
@@ -1758,6 +1758,49 @@ def test_create_test_longname(self):
     ###########################################################################
         self._create_test(["SMS.f19_g16.2000_SATM_XLND_SICE_SOCN_XROF_XGLC_SWAV", "--no-build"])
 
+    ###########################################################################
+    def test_env_loading(self):
+    ###########################################################################
+        self._create_test(["--no-build", "TESTRUNPASS.f19_g16_rx1.A"], test_id=self._baseline_name)
+
+        casedir = os.path.join(self._testroot,
+                               "%s.%s" % (CIME.utils.get_full_test_name("TESTRUNPASS.f19_g16_rx1.A", machine=self._machine, compiler=self._compiler), self._baseline_name))
+        self.assertTrue(os.path.isdir(casedir), msg="Missing casedir '%s'" % casedir)
+
+        with Case(casedir, read_only=True) as case:
+            env_mach = case.get_env("mach_specific")
+            if env_mach.get_module_system_type() == "module":
+                orig_env = dict(os.environ)
+
+                env_mach.load_env(case)
+                module_env = dict(os.environ)
+
+                os.environ.clear()
+                os.environ.update(orig_env)
+
+                env_mach.load_env(case, force_method="generic")
+                generic_env = dict(os.environ)
+
+                os.environ.clear()
+                os.environ.update(orig_env)
+
+                problems = ""
+                for mkey, mval in module_env.items():
```

Comment:
@jedwards4b , I discovered this test is very sensitive to user environment, we may need to add additional exceptions to the environment comparison. I do think a test like this is pretty important as it immediately caught a major bug in generic env loading.